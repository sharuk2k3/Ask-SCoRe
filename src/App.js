import React, {Component} from "react";
import "react-native-paper";
import OnboardingScreen from "./screens/OnboardingScreen/OnboardingScreen";
import StartScreen from "./screens/StartScreen/StartScreen";
import {NavigationContainer} from "@react-navigation/native";
import {createStackNavigator} from "@react-navigation/stack";
import {AsyncStorage} from "react-native";
import {useEffect} from "react";
import config from "./config/config";
import RNBootSplash from "react-native-bootsplash";

const AppStack = createStackNavigator();

class App extends Component {
  state = {isFirstLaunch: null};

  componentDidMount() {
    setTimeout(() => {
      RNBootSplash.hide();
    }, 2000);
    const {isFirstLaunch} = this.state;
    AsyncStorage.getItem(config.ALREADY_LAUNCHED).then(value => {
      if (value == null) {
        AsyncStorage.setItem(config.ALREADY_LAUNCHED, "true");
        this.setState({isFirstLaunch: true});
      } else {
        this.setState({isFirstLaunch: false});
      }
    });
  }

  render() {
    const {isFirstLaunch} = this.state;
    if (isFirstLaunch === null) {
      return null;
    } else if (isFirstLaunch === true) {
      return (
        <NavigationContainer>
          <AppStack.Navigator headerMode="none">
            <AppStack.Screen name="Onboarding" component={OnboardingScreen} />
            <AppStack.Screen name="Start" component={StartScreen} />
          </AppStack.Navigator>
        </NavigationContainer>
      );
    } else {
      return <StartScreen />;
    }
  }
}
export default App;
