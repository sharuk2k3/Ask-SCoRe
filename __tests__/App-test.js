import "react-native";
import React from "react";
import App from "../src/App";
import renderer from "react-test-renderer";

jest.useFakeTimers();

jest.mock("react-native-bootsplash", () => {
  return {
    hide: jest.fn(),
  };
});

test("renders Question Screen correctly", () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});
