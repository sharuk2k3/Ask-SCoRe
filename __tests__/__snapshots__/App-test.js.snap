// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Question Screen correctly 1`] = `null`;
